shared_model/backend/protobuf/commands/proto_add_peer.hpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_ADD_PEER_HPP
+#define IROHA_PROTO_ADD_PEER_HPP
+
+#include <utility>
+
+#include "commands.pb.h"
+#include "interfaces/commands/add_peer.hpp"
+
+namespace shared_model {
+  namespace proto {
+
+    class AddPeer final : public interface::AddPeer {
+     public:
+      template <typename CommandType>
+      explicit AddPeer(CommandType &&command)
```

Comment:
AddPeer can be created from any command type?

Reply:
Assumed that `Command` has the add_peer, look at the load<..> method